main.go

```diff
@@ -1,115 +1,144 @@
+	"github.com/cloudfoundry/noaa/consumer"
 	"github.com/quipo/statsd"
 	"gopkg.in/alecthomas/kingpin.v2"
 )
 
 var (
-	dopplerEndpoint   = kingpin.Flag("doppler-endpoint", "Doppler endpoint").Default("wss://doppler.10.244.0.34.xip.io:443").OverrideDefaultFromEnvar("DOPPLER_ENDPOINT").String()
-	uaaEndpoint       = kingpin.Flag("uaa-endpoint", "UAA endpoint").Default("https://uaa.10.244.0.34.xip.io").OverrideDefaultFromEnvar("UAA_ENDPOINT").String()
-	subscriptionId    = kingpin.Flag("subscription-id", "Id for the subscription.").Default("firehose").OverrideDefaultFromEnvar("SUBSCRIPTION_ID").String()
+	apiEndpoint       = kingpin.Flag("api-endpoint", "API endpoint").Default("https://api.10.244.0.34.xip.io").OverrideDefaultFromEnvar("API_ENDPOINT").String()
 	statsdEndpoint    = kingpin.Flag("statsd-endpoint", "Statsd endpoint").Default("10.244.11.2:8125").OverrideDefaultFromEnvar("STATSD_ENDPOINT").String()
 	statsdPrefix      = kingpin.Flag("statsd-prefix", "Statsd prefix").Default("mycf.").OverrideDefaultFromEnvar("STATSD_PREFIX").String()
-	prefixJob         = kingpin.Flag("prefix-job", "Prefix metric names with job.index").Default("false").OverrideDefaultFromEnvar("PREFIX_JOB").Bool()
-	username          = kingpin.Flag("username", "Firehose username.").Default("admin").OverrideDefaultFromEnvar("FIREHOSE_USERNAME").String()
-	password          = kingpin.Flag("password", "Firehose password.").Default("admin").OverrideDefaultFromEnvar("FIREHOSE_PASSWORD").String()
+	username          = kingpin.Flag("username", "UAA username.").Default("").OverrideDefaultFromEnvar("USERNAME").String()
+	password          = kingpin.Flag("password", "UAA password.").Default("").OverrideDefaultFromEnvar("PASSWORD").String()
 	skipSSLValidation = kingpin.Flag("skip-ssl-validation", "Please don't").Default("false").OverrideDefaultFromEnvar("SKIP_SSL_VALIDATION").Bool()
 	debug             = kingpin.Flag("debug", "Enable debug mode. This disables forwarding to statsd and prints to stdout").Default("false").OverrideDefaultFromEnvar("DEBUG").Bool()
+	updateFrequency   = kingpin.Flag("update-frequency", "The time in seconds, that takes between each apps update call.").Default("300").OverrideDefaultFromEnvar("UPDATE_FREQUENCY").Int64()
+	metricTemplate    = kingpin.Flag("metric-template", "The template that will form a new metric namespace.").Default("{{.Space}}.{{.App}}.{{.Metric}}").OverrideDefaultFromEnvar("METRIC_TEMPLATE").String()
 )
 
 func main() {
 	kingpin.Parse()
 
-	tokenFetcher := &token.UAATokenFetcher{
-		UaaUrl:                *uaaEndpoint,
-		Username:              *username,
-		Password:              *password,
-		InsecureSSLSkipVerify: *skipSSLValidation,
+	c := &cfclient.Config{
+		ApiAddress:        *apiEndpoint,
+		SkipSslValidation: *skipSSLValidation,
+		Username:          *username,
+		Password:          *password,
 	}
 
-	authToken, err := tokenFetcher.FetchAuthToken()
+	client, err := cfclient.NewClient(c)
```

Comment:
I don't know, if we can. I would of think, this constant is set on the API server side and we won't have the access to it, as it would allow people to change it into years.

But, from my understanding, we should never face that issue, and here's why:

1. We're running `routine` every (let's say) `30 seconds`.
2. The `AccessToken` is expired after `10 minutes`.
3. The `cfclient` library calls into the API requesting new `AccessToken` and authenticates with the existing `RefreshToken`.
4. API issues new `AccessToken` and _SHOULD_ issue new `RefreshToken` as well.
  - I say should, as otherwise anyone that gets a handle of that `RefreshToken`, essentially gathers the access to new `AccessToken` instantly...
5. New `RefreshToken` issued every 10 minutes prevents the expiry scheduled for (let's say) 7 days.

It may be the case, where the authorising server is configured to extend the lifespan of existing token as oppose to generating new one.

There may be a concern, that the UAA doesn't actually do that - Should we perhaps confirm the above workflow?

I _think_, the issue Dan is referring to, is somewhat unrelated to our use case.